cpp/ql/src/Security/CWE/CWE-676/PotentiallyDangerousFunction.ql

```diff
@@ -7,18 +7,23 @@
  * @id cpp/potentially-dangerous-function
  * @tags reliability
  *       security
- *       external/cwe/cwe-676
+ *       external/cwe/cwe-242
  */
 import cpp
 
-
-predicate dangerousFunction(Function function) {
-  exists (string name | name = function.getQualifiedName() |
-    name = "gmtime")
+predicate potentiallyDangerousFunction(Function f, string message) {
+  (
+    f.getQualifiedName() = "gmtime" and
+    message = "Call to gmtime is potentially dangerous"
+  ) or (
+    f.hasName("gets") and
+    message = "gets does not guard against buffer overflow"
```

Comment:
It's possible that use of `gets` should have severity "error", while use of `gmtime` should just be "warning". But this PR is a step in the right direction, and I'm happy to merge it without severity changes.

Reply:
Yes, that's probably what I'd have said as well, but I don't have a strong reason to change it.

There was at one point talk of allowing precision and severity to be a query output, which would be great for situations like this.  I'm not aware this feature is on anyone's short term plans though.

Reply:
IIRC I brought it up around Christmas in a product planning setting, but there was no appetite for it.